examples/http3-server.rs

```diff
@@ -636,7 +638,11 @@ fn handle_writable(client: &mut Client, stream_id: u64) {
     let written = match http3_conn.send_body(conn, stream_id, body, true) {
         Ok(v) => v,
 
+        Err(quiche::h3::Error::Done) => 0,
+
         Err(e) => {
+            client.partial_responses.remove(&stream_id);
```

Comment:
I was confused about this actually, but AFAICT `StreamBlocked` is only returned by `send_request()`, `send_response()` and `send_goaway()`, while `send_body()` returns `Done` (hence why it is already handled separately).

Which is not that great, but for the purposes of this PR it's fine-ish I guess...

Reply:
Yeah, you're right. The `send_body()` docs hint at this but it wouldn't hurt to day that we'll never return StreamBlocked. And that we could return StreamStopped?

Reply:
Currently `StreamStopped` is a QUIC-only error, so in HTTP/3 it would be converted into `h3::Error::TransportError(Error::StreamStopped(...))` (and returning `TransportError` in `send_body()` was already possible before I think, though not documented).

However as I suggested in the commit message, I think what we could do is turn `StreamStopped` into HTTP/3-specific errors, so for example `Error::StreamStopped(0x010c)` could be cnvrted into `h3::Error::RequestCancelled`, and so forth, so the application doesn't need to do this on its own. Though it wasn't immediately clear to me what to do in some cases, e.g. when `H3_NO_ERROR` is received, so I left this to a future PR to avoid getting bogged down on that.

I think we can overhaul the `send_body()` docs once we have decided what to do here (but I agree that they could be better).

Reply:
Future change works for me